src/backend.rs

```diff
@@ -182,10 +198,14 @@ where
                     let remaining = available - amt;
                     remainders.push((remaining, address.clone(), policy.clone()));
                 } else {
-                    return Err(format!("Not enough {:?}", policy));
+                    return Err(Error::TxORecord(TxORecordError::InsufficientAmountOf(
```

Comment:
Ah. My bad. These should just be core errors here, since they aren't behind any interfaces. This is the true domain code.

You can just move the `InsufficientAmountOf` variant to the core error enum.